net/ipv6: allow device only routes via the multipath API

This reverts commit b5d2d75e079a918be686957b1a8d2f6c5cc95a0a
net/ipv6: Do not allow device only routes via the multipath API

The main reason for that commit was because a mixture of device
only and gateway address nexthops for ipv6 is broken.
However, our CLI does not allow this mixture to be configured.
So allow multiple device only nexthops again to restore the
previous functionality.

Signed-off-by: George Wilkie <gwilkie@vyatta.att-mail.com>
---
 net/ipv6/route.c |    7 -------
 1 file changed, 7 deletions(-)

--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -4417,13 +4417,6 @@ static int ip6_route_multipath_add(struc
 			rt = NULL;
 			goto cleanup;
 		}
-		if (!rt6_qualify_for_ecmp(rt)) {
-			err = -EINVAL;
-			NL_SET_ERR_MSG(extack,
-				       "Device only routes can not be added for IPv6 using the multipath API.");
-			fib6_info_release(rt);
-			goto cleanup;
-		}
 
 		rt->fib6_nh.nh_weight = rtnh->rtnh_hops + 1;
 
